### PR TITLE
refactor(bulk_load): refactor get_app for bulk_load_service

### PR DESCRIPTION
**Pull request by hycdong**
This pull request adds `get_app` function for class bulk_load_service

### CODE REVIEW

File: src/meta/meta_bulk_load_service.h [hunk 1]
```diff
@@ -290,6 +290,18 @@ class bulk_load_service
...
+    inline std::shared_ptr<app_state> get_app(const std::string &name)
+    {
+        zauto_read_lock l(app_lock());
+        return _state->get_app(name);
+    }
+
+    inline std::shared_ptr<app_state> get_app(int32_t app_id)
+    {
+        zauto_read_lock l(app_lock());
+        return _state->get_app(app_id);
```
Comment (levy5307) [inline src/meta/meta_bulk_load_service.h:302]: The lifecycle is maintained by _state, so I think there is no need to return std::shared_ptr here

File: src/meta/meta_bulk_load_service.cpp [hunk 2]
```diff
@@ -66,25 +66,20 @@ void bulk_load_service::on_start_bulk_load(start_bulk_load_rpc rpc)
...
+        auto hint = fmt::format(
+            "app {} is ", response.err == ERR_APP_NOT_EXIST ? "not existed" : "not available");
+        response.hint_msg = hint;
+        derror_f("{}", hint);
+        return;
+    }
+    if (app->is_bulk_loading) {
+        derror_f("app({}) is already executing bulk load, please wait", app->app_name);
+        response.err = ERR_BUSY;
+        response.hint_msg = "app is already executing bulk load";
```
Comment (Shuo-Jia) [inline src/meta/meta_bulk_load_service.cpp:81]: like L72, define hint_msg and reuse it to derror_f(hint_msg)

Comment (hycdong) [reply src/meta/meta_bulk_load_service.cpp]: Done

File: src/meta/meta_bulk_load_service.cpp [hunk 3]
```diff
@@ -66,25 +66,21 @@ void bulk_load_service::on_start_bulk_load(start_bulk_load_rpc rpc)
...
+    if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+        response.err = app == nullptr ? ERR_APP_NOT_EXIST : ERR_APP_DROPPED;
+        auto hint = fmt::format(
+            "app {} is ", response.err == ERR_APP_NOT_EXIST ? "not existed" : "not available");
+        response.hint_msg = hint;
+        derror_f("{}", hint);
+        return;
+    }
+    if (app->is_bulk_loading) {
+        auto hint = fmt::format("app({}) is already executing bulk load", app->app_name);
```
Comment (levy5307) [inline src/meta/meta_bulk_load_service.cpp:79]:
```
        response.hint_msg = fmt::format("app({}) is already executing bulk load", app->app_name);
        response.err = ERR_BUSY;
        derror_f("{}", response.hint_msg);
```
avoid to create another string var

File: src/meta/meta_bulk_load_service.cpp [hunk 4]
```diff
@@ -1380,34 +1355,28 @@ void bulk_load_service::on_query_bulk_load_status(query_bulk_load_rpc rpc)
...
-        if (!app->is_bulk_loading) {
-            auto hint_msg = fmt::format("app({}) is not during bulk load", app_name);
-            derror_f("{}", hint_msg);
-            response.err = ERR_INVALID_STATE;
-            response.__set_hint_msg(hint_msg);
-            return;
-        }
+    std::shared_ptr<app_state> app = get_app(app_name);
+    if (app == nullptr || app->status != app_status::AS_AVAILABLE) {
+        auto hint_msg = fmt::format("app({}) is not existed or not available", app_name);
```
Comment (levy5307) [inline src/meta/meta_bulk_load_service.cpp:1360]: same

**Review by levy5307 — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**